### PR TITLE
create Api folder with one controller and request test

### PR DESCRIPTION
**Pull request by aydamacink**
-  +gem 'responders', '~> 2.0'

I accidentally commited a file who does not need to be here so I'll correct in the next commit:
spec/spec-requests/drop_requests.rb

The next step is to make the place comply with json  api specs

### CODE REVIEW

**Comment by aydamacink** [comment]
I deleted the file

**Comment by siemieniec** [comment]
Good work. I didn't know that responders where extracted. Need to keep up changes in Rails.
- I would delete those empty files:
  
  ```
  app/assets/javascripts/api/drops.coffee
  app/assets/stylesheets/api/drops.scss
  app/helpers/api/drops_helper.rb
  spec/helpers/api/drops_helper_spec.rb 
  ```
  
  They will probably never be used and should not be added to the project.
- And I would move rather move tests from
  `spec/requests/api/drops_controller_spec.rb`
  to
  `spec/controllers/api/drops_controller_spec.rb`
  or move current tests from `spec/controllers` to `spec/requests`.
  They have same form and are testing exactly same things. I would prefer keeping those type of tests in a request directory but maybe you should follow project structure.

File: spec/requests/api/drops_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+
+RSpec.describe Api::DropsController, :vcr => {:cassette_name => "place" }, type: :request do
```
Comment (siemieniec) [inline spec/requests/api/drops_controller_spec.rb]: You should change the name of cassette. `place` is already used in the other test so this call will overwrite it.

Comment (aydamacink) [inline spec/requests/api/drops_controller_spec.rb]: Thanks for the comments ... can I change the name to anything at all? or what should the name be?

Comment (siemieniec) [inline spec/requests/api/drops_controller_spec.rb]: According to me the best name is no name. :-)
If you label the test like this: `RSpec.describe Api::DropsController, :vcr, type: :request do` VCR will automagically create cassette in `spec/fixtures/[describe_block_name]/[test_block_name]`.

**Comment by langjoh** [comment]
@starkcoffee Regarding Robert's suggestion of moving the tests: What do you think? We thought the difference between request and controller specs should be mirrored in the folder structure?

File: spec/requests/api/drops_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+
+RSpec.describe Api::DropsController, :vcr, type: :request do
+  context 'with several drops present' do
+  let!(:drops) { FactoryGirl.create_list(:drop, 3) }
+
+    describe "GET /drops" do
+
+      it "displays list of  drops in database" do
+
```
Comment (rstammer) [inline spec/requests/api/drops_controller_spec.rb]: Just a minor one: I'd suggest dropping the unnecessary whitespace lines (L2 and L11) to keep the house clean.

File: app/views/api/drops/index.json.jbuilder [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  json.title drop.title
+  json.sc_track drop.sc_track
+  json.place do
+    json.id drop.place.id
+    json.name drop.place.name
+    json.latitude drop.place.latitude
+    json.longitude drop.place.longitude
+  end
+end
+
```
Comment (rstammer) [inline app/views/api/drops/index.json.jbuilder]: Same story here: As long as whitespace has the function of improving readability I would keep them, but I'd suggest dropping the whitespace line at the end here.

File: spec/controllers/api/drops_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Api::DropsController, type: :controller do
+
```
Comment (rstammer) [inline spec/controllers/api/drops_controller_spec.rb]: This controller test is currently empty and therefore "adds no value" to the codebase. Since you have proper request specs, I tend to suggest dropping it entirely. But you could also consider filling it with examples, if you like.

**Comment by starkcoffee** [comment]
@Powerglove What is the difference between request and controller tests? I haven't heard of request tests before. I would have put them in the `controller` directory, but if `requests` is the new way of doing things, fine with that too.

**Comment by starkcoffee** [comment]
Do we know why there are all these cassette files? Not sure why there are cassette files which contain Google geocoding requests? Do the tests work if we don't have those cassette files?

Great work! 🔥

**Comment by aydamacink** [comment]
@starkcoffee :  Request tests don't test implementation details of the controller, but dogmatically the input , the output and implicitly the routes. They  also  test the controller - but only the request that the controller gets- and the templates  or output (for example redirects_to),  whereas controller tests focus on testing implementation details (That's why they make use of  instance variables). In our  humble opinion, it wouldn't make much sense to put them in the controller directory.

**Comment by siemieniec** [comment]
@aydamacink @starkcoffee There is no strong line between request and controller tests. In fact with Rails 5 official recommendation is to use controller tests as @aydamacink described request tests.
You can read more about it here: http://blog.bigbinary.com/2016/04/19/changes-to-test-controllers-in-rails-5.html#reasons-for-removing-assigns-and-asserttemplate

**Comment by berlintam** [comment]
amazing
